src/pages/AddData.tsx

```diff
@@ -0,0 +1,36 @@
+import React from 'react';
+import { withOidcSecure } from '@axa-fr/react-oidc';
+import { useBaseFields } from '../pdc-api';
+import { PanelGrid } from '../components/PanelGrid';
+import { BaseFields } from '../components/BaseFields';
+import { BulkUploader } from '../components/BulkUploader';
+
+const BulkUploaderLoader = () => {
+  const history: object[] = []; // todo
+  return (
+    <BulkUploader history={history} loading />
+  );
+};
+
+const BaseFieldsLoader = () => {
+  const fields = useBaseFields();
+  if (fields === null) {
+    return (
+      <BaseFields fields={[]} loading />
+    );
+  }
+
+  return (
+    <BaseFields fields={fields} />
+  );
+};
+
+const AddDataLoader = () => (
+  <PanelGrid sidebarred>
+    <BulkUploaderLoader />
+    <BaseFieldsLoader />
```

Comment:
Two questions:

1. I was hoping you'd go ahead and generate the panels for these as part of the task (though perhaps not part of this PR, if you're multi-staging it). My apologies, that was only lightly implied through my linking to the [Panel Grid Plus Sidebar](https://pilot.philanthropydatacommons.org/storybook/?path=/story/stories-panelgrid--plus-sidebar) component in the task description. Do you plan to deliver that as a separate PR, or does it make sense as this one? (The content can still be all "TODO"!)
2. The main (right-hand) panel is actually more than just base fields description, it's the "instructional text panel". The base field table is the most important bulk of it, but I think the hierarchy should be an instructional panel at the top with the base field table as just one of the content children. With that in mind, I think perhaps we need a different component here, which then internally renders `<BaseFieldsLoader />`?

Reply:
…or perhaps all that's needed to rename `BaseFieldsLoader` to `AddDataInstructionPanelLoader` or something? I assume we want the API call to be as close to the relevant child as possible. Does that mean the panel makes the call, or just the section of the panel that contains the base fields?

Reply:
> I was hoping you'd go ahead and generate the panels for these

Ah sure, I can handle that as part of this PR!

> With that in mind, I think perhaps we need a different component here, which then internally renders `<BaseFieldsLoader />`?

> …or perhaps all that's needed to rename `BaseFieldsLoader` to `AddDataInstructionPanelLoader` or something? I assume we want the API call to be as close to the relevant child as possible. Does that mean the panel makes the call, or just the section of the panel that contains the base fields?

The approach I've been taking - and we can certainly revisit it - is to separate all of the API calls out into the `<...Loader>`s which live in the `src/pages/` directory. Under that model, we'd pass in the base fields to the component that renders it. That allows us to use Storybook more easily, and hopefully makes it easier for you to create components as needed!

Perhaps these components should have a parent-child relationship?

```jsx
<AddDataInstructionPanel>
  <BaseFieldsLoader />
</AddDataInstructionPanel>
```

Reply:
Alternatively, they could be peers, and maybe we could even have some visual separation between the instructions and the base field list? Although we don't have any pages with multiple panels in a vertical alignment, and in fact displaying multiple `<Panel>`s inside a single `<PanelGridItem>` causes them to stack on top of each other in a last-one-wins scenario.

What do you think, @reefdog?

Reply:
```jsx
<AddDataInstructionPanel>
  <BaseFieldsLoader />
</AddDataInstructionPanel>
```

This is exactly how I was envisioning it! Albeit eventually with a sibling component to `<BaseFieldsLoader />` that contained the instructional text. Referencing the image:

![Upload CSV](https://github.com/PhilanthropyDataCommons/front-end/assets/4731/da448919-e554-4a77-9c95-22cc2c2c938c)

We'd end up with some static content above the BaseFieldsLoader, e.g. (in this very much pseudo-markup):

```
<AddDataInstructionPanel>
  <h1>Add data to the PDC</h1>
  <p>Adding data via the API…</p>
  <p>Adding data via the bulk uploader…</p>
  <h2>Base Fields</h2>
  <p>Base fields are the backbone…</p>
  <BaseFieldsLoader />
</AddDataInstructionPanel>
```

With the exception that maybe the base field static content would also be within the loader, I don't know. But I think you see what i mean here?

Reply:
Does that fit your model, too?

Reply:
That makes sense! I've updated the branch with these changes. Let me know if this is closer to what you need!